embedded/sql/catalog.go

```diff
@@ -232,6 +234,10 @@ func (i *Index) IsPrimary() bool {
 	return i.id == PKIndexID
 }
 
+func (i *Index) IsAutoIncrement() bool {
```

Comment:
I'm wondering if it's the autoincrement concept or feature belong to the index... It may be of the column but the index may be needed in order to speed up fetching the greatest value and ensuring uniqueness, but it should be technically possible to have an autoincremental column even without a real index (yet unpractical, but just so assign "responsibilities")

Reply:
can you explain in more detail?

Reply:
that I wouldn't assign to the index the notion of "autoincrement"... but on the logic implementing autoincrement search for the index that can be used for it... it's not a functional but implementation comment

Reply:
Snap theif been long time sorry...